CorsixTH/Src/th_pathfind.cpp

```diff
@@ -290,9 +290,15 @@ bool idle_tile_finder::find_idle_tile(const level_map* pMap, int iStartX,
 
   while (true) {
     pNode->visited = true;
-    map_tile_flags flags = pMap->get_tile_unchecked(pNode->x, pNode->y)->flags;
 
-    if (!flags.do_not_idle && flags.passable && flags.hospital) {
+    // Check whether pNode is a feasible destination.
+    const map_tile* node_tile = pMap->get_tile_unchecked(pNode->x, pNode->y);
+    map_tile_flags flags = node_tile->flags;
+
+    bool correct_parcel = parcelId <= 0 || parcelId == node_tile->iParcelId;
```

Comment:
Confused about whether we're suppose to be respecting parcel 0 or not.

Reply:
Originally, I thought parcel `0` was the first used parcel, so I picked `-1` for not having a parcel requirement. Afterwards, mugmuggy wanted parcel `0` out, as it turned out to be outside, which is not where staff idles. Changing `< 0` to `<= 0` was the least amount of work at that point, so that's why we have the current implementation.

We do rely on Lua doing sane things here, as we neither check existence of a parcel number in CPP nor do we check if the given parcel is owned by the hospital of the staff.

We seem to have 3 useful solutions.
- Keep it as-is, with the advantage that the "lack of parcel requirement" is very clearly not a parcel.
- Also allow parcel `0` requirement so if we ever want a path to outside and current methods are not sufficient, it can be computed (ie revert back to `< 0`).
- Use `0` as lack of parcel since it's unlikely we'll ever need it to navigate to the outdoor-parcel. It's somewhat fuzzy that parcel `0` then has two meanings depending on what code you are looking at.

I have no strong feelings about any of the solutions, although the 3rd one looks less nice to me.

Reply:
The parcel < 0 and parcel <= 0 thing came because we had code to not perform it when it was 0, if you remove that check you get incorrect behaviour as its false in certain circumstances (0 < 0)' || '0 == n (as n > 0 inside the hospital typically)):
Line 72 in meander.lua to protect the cpp call from scenarios that introduce that false.
```lua
if humanoid.humanoid_class == "Handyman" and humanoid.parcelNr > 0 and
```
if you ignore the room check on that if, the meander can pass the parcelNr as the last argument.
```lua
      @param parcelId Id of the parcel with valid destinations, use \c -1 for
     any parcel.
```
So we use 0 for any plot on the staff member and -1 in cpp (new) as Albert points out but just to correct Albert on the outside parcel, as I pointed out, hospital tiles outside the hospital walls as parcelNr is 0, pull the handyman outside when the parcelNr check is removed (https://github.com/CorsixTH/CorsixTH/issues/2014#issuecomment-887968306 for a reference of the map issue and also #1991 ). They just don't meander outside anywhere. Where the map has no external hospital tiles, and the check is removed and you place a handyman (say L2 map) it crashes as there is no valid destination tile.

Reply:
AFAIK using 0 for not in hospital is problematic, many maps stretch the parcel numbers well outside of the bounds of the hospital, and doing so has a real world use - to make the cost of some parcels larger than what would normally be calculated from the hospital.

I don't think there's any valid reason now or ever to query for parcel 0 when path finding, and suggest changing the default to 0 in cpp?

Reply:
Parcel 0 it is.

I did wonder about modifying maps after loading to clean up errors, but never tried to implement it.

As for parcel areas outside the hospital, could be nice to have an outdoor ward :p  
A more real use I guess is to artificially increase the parcel cost, which is arguably better solved by adding such a parameter to the level file of course.